src/nifreeze/model/dmri.py

```diff
@@ -209,21 +159,21 @@ def predict(self, gradient=None, **kwargs):
 
             predicted = np.hstack(predicted)
 
-        if self._mask is not None:
-            retval = np.zeros_like(self._mask, dtype="float32")
-            retval[self._mask, ...] = predicted
+        if brainmask is not None:
+            retval = np.zeros_like(brainmask, dtype="float32")
+            retval[brainmask, ...] = predicted
         else:
-            retval = predicted.reshape(self._datashape[:-1])
+            retval = predicted.reshape(self._dataset.shape[:-1])
 
         return retval
 
 
-class AverageDWIModel(BaseDWIModel):
+class AverageDWIModel(ExpectationModel):
```

Comment:
I held up changing Average here until I got a sense of how you all receive "Expectation". The final merge should have consistent naming.